test/ecpair.js

```diff
@@ -45,61 +47,68 @@ describe('ECPair', function () {
     })
 
     fixtures.valid.forEach(function (f) {
-      it('calculates the public point for ' + f.WIF, function () {
-        var d = new BigInteger(f.d)
-        var keyPair = new ECPair(d, null, {
+      it('derives public key for ' + f.WIF, function () {
+        let d = Buffer.from(f.d, 'hex')
+        console.log(d)
+
+        let keyPair = ECPair.fromPrivateKey(d, {
           compressed: f.compressed
         })
 
-        assert.strictEqual(keyPair.getPublicKeyBuffer().toString('hex'), f.Q)
+        assert.strictEqual(keyPair.publicKey.toString('hex'), f.Q)
       })
     })
 
     fixtures.invalid.constructor.forEach(function (f) {
       it('throws ' + f.exception, function () {
-        var d = f.d && new BigInteger(f.d)
-        var Q = f.Q && ecurve.Point.decodeFrom(curve, Buffer.from(f.Q, 'hex'))
-
-        assert.throws(function () {
-          new ECPair(d, Q, f.options)
-        }, new RegExp(f.exception))
+        if (f.d) {
+          let d = Buffer.from(f.d, 'hex')
+          assert.throws(function () {
+            ECPair.fromPrivateKey(d, f.options)
+          }, new RegExp(f.exception))
+        } else {
+          let Q = Buffer.from(f.Q, 'hex')
+          assert.throws(function () {
+            ECPair.fromPublicKey(Q, f.options)
+          }, new RegExp(f.exception))
+        }
       })
     })
   })
 
-  describe('getPublicKeyBuffer', function () {
-    var keyPair
+  describe('getPublicKey', function () {
+    let keyPair
 
     beforeEach(function () {
-      keyPair = new ECPair(BigInteger.ONE)
+      keyPair = ECPair.fromPrivateKey(ONE)
     })
 
-    it('wraps Q.getEncoded', hoodwink(function () {
-      this.mock(keyPair.Q, 'getEncoded', function (compressed) {
-        assert.strictEqual(compressed, keyPair.compressed)
-      }, 1)
+    it('calls pointFromScalar lazily', hoodwink(function () {
+      assert.strictEqual(keyPair.__Q, null)
 
-      keyPair.getPublicKeyBuffer()
+      // .publicKey forces the memoization
+      assert.strictEqual(keyPair.publicKey.toString('hex'), '0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798')
+      assert.strictEqual(keyPair.__Q.toString('hex'), '0279be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798')
     }))
   })
 
   describe('fromWIF', function () {
     fixtures.valid.forEach(function (f) {
       it('imports ' + f.WIF + ' (' + f.network + ')', function () {
-        var network = NETWORKS[f.network]
-        var keyPair = ECPair.fromWIF(f.WIF, network)
+        let network = NETWORKS[f.network]
+        let keyPair = ECPair.fromWIF(f.WIF, network)
 
-        assert.strictEqual(keyPair.d.toString(), f.d)
+        assert.strictEqual(keyPair.privateKey.toString('hex'), f.d)
```

Comment:
The hex will not be padded so it may contain fewer than 64 characters for some private keys, such as this LTC private key: T34MuTEaRxt3kj6dn2MMHbvSPQHrsBjKrdz6WtwrLe1KzCzYkMup.

Shouldn't it be padded to conform to 64 characters?

Reply:
@woodser `.privateKey` is enforced to be 256-bits, or 32-bytes in `.length`.